source/config/__init__.py

```diff
@@ -628,6 +630,11 @@ def deleteProfile(self, name):
 				if trigger._profile == delProfile:
 					del self._suspendedTriggers[trigger]
 
+	def getActiveProfile(self):
+		if globalVars.appArgs.secure:

```

Comment:
I don't think this should return a different value when in secure mode - while it is indeed impossible to create  profiles or triggers when running securely triggers created in the configuration before it is copied to the system config are still applied.

Reply:
What do you suggest?

My understanding is that if globalVars.appArgs.secure is on, then no custom profiles are allowed, so we return None to signalize that the default profile is used.

Reply:
> My understanding is that if globalVars.appArgs.secure is on, then no custom profiles are allowed, 

That is not true. You can test as follows:
1. Create a triggered profile for an application and change some settings there.
2. Start NVDA providing `--secure` from the CLI.
3. Check if the trigger for the configure profile has been applied or not.

> What do you suggest?

This method should not do anything special when running with `globalVars.appArgs.secure` set to `true.

Reply:
@feerrenrut  Can you explain what globalVars.appArgs.secure  does exactly?

@lukaszgo1  while I agree with you, in several places where globalVars.appArgs.secure is checked things like saving profiles seem to be disabled.

See, for example, the constructor of class ConfigManager in config/__init__.py 

You will find the following piece of code:

```
self._shouldWriteProfile: bool = not (globalVars.appArgs.secure or globalVars.appArgs.launcher)
```

If I cannot save a profile, how am I supposed to switch to it?

Again, I would like to know what does this variable do. I couldn't find documentation.

Reply:
From the [user guide, advanced topics, command line options](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#CommandLineOptions):
`--secure`: Secure mode: disables Python console, profile features such as creation, deletion, renaming profiles etc., update check, some checkboxes in the welcome dialog and in general settings category (e.g. Use NVDA during sign-in, save configuration after exit etc.), as well as logviewer and logging features (used often in secure screens). Note also that this command will disable the possibility to save settings in system config and the gesture map will not be saved on the disk.

The primary use case is when NVDA is run as the "System" user, eg during sign-on. Then the secure argument is set, search for `isSecureDesktop = desktopName == "Winlogon"` in `nvda.pyw`.

Reply:
My bad. I should have looked first in the user guide instead ratter than spending more than 20 minutes inspecting the source code.

@lukaszgo1  thanks so much for pointing this out, it is now fixed and hopefully folks won a more elegant way than referencing [-1] every where.

@feerrenrut  if there is anything else remaining let me know. Also, feel free to edit or ask some one to edit the documentation, I have done nearly my best but English is not my native language.